lib/iwdp.js

```diff
@@ -0,0 +1,81 @@
+import { exec, SubProcess } from 'teen_process';
+import EventEmitter from 'events';
+import logger from './logger';
+import request from 'request-promise';
+import { retryInterval } from 'asyncbox';
+
+const IWDP_CMD = 'ios_webkit_debug_proxy';
+const MAX_RETRIES = 10;
+
+class IWDP extends EventEmitter {
+
+  constructor (udid = null) {
+    super();
+    this.udid = udid;
+    this.attempts = 0;
+    this.port = 27753;
+    this.process = this.createIWDPProcess(); 
+    this.endpoint = `http://localhost:${this.port}`;
+  }
+
+  createIWDPProcess () {
+    let process = new SubProcess(IWDP_CMD, ['--config', `${this.udid}:${this.port},:${this.port + 1}-${this.port + 101}`]); // (see https://github.com/google/ios-webkit-debug-proxy for reference)// If the process exits and the exit wasn't requested by the API, restart it
+    process.on('exit', () => this.onExit());
+    return process;
+  }
+
+  async onExit () {
+    if (!this.exitRequested) {
+      this.process = this.createIWDPProcess();
+      await this.start(); 
+    }
+  }
+
+  async start () {
+    if (++this.attempts > MAX_RETRIES) {
```

Comment:
When the process exits, `onExit` is called. If the 'exit' is unexpected (ie: it wasn't called in iwdp.stop method) then we call start again (`await this.start()`). If it does this 10 times then it emits an 'error' event and stops retrying.